### ISSUE TITLE
Triangle Inequality exercuse diverges from core problem specification & get too complicated

### ISSUE THREAD

**Issue by dhavalgshah** [exercism/go#1477]
There three issues with Triangle classification exercise:

1. Test cases for `divergent triangles`: Per [problem-specifications#202](https://github.com/exercism/problem-specifications/issues/202) they should have been removed because of excess complexity they add. Further, README says _Feel free to add your own code/tests to check for degenerate triangles._ which implies that is an advance use case and is optional.
2. Test cases with `Inf+` lengths: Neither common problem specification has any such test cases nor many other tracks that I could check. Even if we are okay in diverging from others, should Infinity comparison be this early on?

3. Confusing README
>For a shape to be a triangle at all, all sides have to be of length > 0, and
the sum of the lengths of any two sides must be greater than or equal to the
length of the third side. See [Triangle Inequality](https://en.wikipedia.org/wiki/Triangle_inequality).

This implies of `z <= x + y` where `z` is the longest side. This also includes `divergent triangles` by definition.

However, just below it in _Dig Deeper_ section it says this:
>The case where the sum of the lengths of two sides _equals_ that of the
third is known as a _degenerate_ triangle - it has zero area and looks like
a single line. **Feel free to add your own code/tests to check for degenerate triangles.** (emphasis added)

---

When you couple all of above, this exercise becomes too complex too fast. One has to think about finding largest side, guard against infinite lengths, go against correct definition of triangle inequality, and then some inputs having overlapping classifications.

IMHO this is a disservice to the learner. Because if I'm unaware about the subject and I read Wikipedia or other material then I would be very confused of why the code based on given theorem is not working. 

It will also force me to think too much about geometry (and algebra) rather than programming because given problem statement is incomplete and relies on too many implied/conflicting statements.

**Comment by jmrunkle** [comment]
I agree, the problem has gotten much more complicated than the problem-specification source and should be improved upon. Are you volunteering or asking if someone else will take care of it?
